File: BLL/Models/AnswerModels/UpsertAnswerModel.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace BLL.Models.AnswerModels
{
    public class UpsertAnswerModel
    {
        public int TestResultId { get; set; }
        public string UserAnswer { get; set; }
        public int QuestionId { get; set; }
        public int? OptionId { get; set; }
    }
}

File: QuizUI/src/app/components/quiz/entities/answer/upsert-answer.model.ts
export class UpsertAnswerModel {
  testResultId: number;
  userAnswer: string;
  questionId: number;
  optionId: number;

  constructor(userAnswer: string, testResultId: number, questionId: number, optionId?: number) {
    this.userAnswer = userAnswer;
    this.optionId = optionId;
    this.questionId = questionId;
    this.testResultId = testResultId;
  }
}
